processes/consumer/flush.go

```diff
@@ -85,35 +89,46 @@ func Flush(ctx context.Context, inMemDB *models.DatabaseData, dest destination.B
 				"reason":   args.Reason,
 			}
 
-			var err error
-			action := "merge"
-			// Merge or Append depending on the mode.
-			if _tableData.Mode() == config.History {
-				err = dest.Append(_tableData.TableData, false)
-				action = "append"
-			} else {
-				err = dest.Merge(_tableData.TableData)
-			}
+			err = retry.WithRetries(retryCfg, func(_ int, _ error) error {
+				return flush(ctx, dest, _tableName, _tableData)
+			})
 
 			if err != nil {
 				tags["what"] = "merge_fail"
-				tags["retryable"] = fmt.Sprint(dest.IsRetryableError(err))
-				slog.With(logFields...).Error(fmt.Sprintf("Failed to execute %s, not going to flush memory, will sleep for 3 seconds before continuing...", action), slog.Any("err", err))
-				time.Sleep(3 * time.Second)
+				slog.Error(fmt.Sprintf("Failed to %s", action), slog.Any("err", err), slog.String("tableName", _tableName))
 			} else {
-				slog.Info(fmt.Sprintf("%s success, clearing memory...", stringutil.CapitalizeFirstLetter(action)), logFields...)
-				commitErr := commitOffset(ctx, _tableData.TopicConfig().Topic, _tableData.PartitionsToLastMessage)
-				if commitErr == nil {
-					inMemDB.ClearTableConfig(_tableName)
-				} else {
-					tags["what"] = "commit_fail"
```

Comment:
If we are permanently unable to commit the offsets to Kafka due to either a misconfiguration or a bug in the code, are we going to get notified about it since we're not logging `commit_fail` anymore?

Reply:
Is there a risk that if we fail to commit offsets for an extended period of time when transfer eventually restarts it will have to pick up from an offset that is significantly in the past?

Reply:
> If we are permanently unable to commit the Kafka offsets due to either a misconfiguration or a bug in the code, are we going to get notified about it if we're not reporting commit_fail metrics anymore?

We still can since warnings go into Datadog too. We could set up an alert there if we wanted.

> Is there a risk that if we fail to commit offsets for an extended period of time when transfer eventually restarts it will have to pick up from an offset that is significantly in the past?

Hm, it is possible, although I haven't seen it before. (We average ~1-3 occasional commit failures per month).

Reply:
I like it, preserving the existing behavior seems like the safer approach to me.